Admin/ArticleAdmin.php

```diff
@@ -0,0 +1,266 @@
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Sonata\AdminBundle\Admin\AbstractAdmin;
+use Sonata\AdminBundle\Admin\AdminInterface;
+use Sonata\AdminBundle\Datagrid\ListMapper;
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\AdminBundle\Show\ShowMapper;
+use Sonata\ArticleBundle\Model\Article;
+use Sonata\ArticleBundle\Model\ArticleInterface;
+use Sonata\ArticleBundle\Model\Fragment;
+use Sonata\CoreBundle\Validator\ErrorElement;
+
+/**
+ * @author Florent Denis <florent.denis@ekino.com>
+ */
+class ArticleAdmin extends AbstractAdmin
+{
+    /**
+     * @var array
+     */
+    protected $datagridValues = array('_sort_by' => 'updatedAt', '_sort_order' => 'DESC');
+
+    /**
+     * @var string
+     */
+    protected $fragmentClass;
+
+    /**
+     * @param string $code
+     * @param string $class
+     * @param string $baseControllerName
+     * @param string $fragmentClass
+     */
+    public function __construct($code, $class, $baseControllerName, $fragmentClass)
+    {
+        parent::__construct($code, $class, $baseControllerName);
+
+        $this->fragmentClass = $fragmentClass;
+    }
+
+    /**
+     * Inspired from Twig_Extensions_Extension_Text.
+     *
+     * {@inheritdoc}
+     */
+    public function toString($object)
+    {
+        $value = $object->getTitle();
+        $length = 80;
+        $separator = '...';
+
+        if (mb_strlen($value) > $length) {
+            // If breakpoint is on the last word, return the value without separator.
+            if (false === ($breakpoint = mb_strpos($value, ' ', $length))) {
+                return $value;
+            }
+
+            $length = $breakpoint;
+
+            return rtrim(mb_substr($value, 0, $length)).$separator;
+        }
+
+        return $value;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTemplate($name)
+    {
+        if ('edit' === $name) {
+            return 'SonataArticleBundle:FragmentAdmin:edit_article.html.twig';
+        }
+
+        return parent::getTemplate($name);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function validate(ErrorElement $errorElement, $object)
+    {
+        $errorElement
+            ->with('title')
+                ->assertNotNull()
+            ->end()
+            ->with('status')
+                ->assertChoice(array_keys($this->isGranted('ROLE_ARTICLE_PUBLISH') ? Article::getStatuses() : Article::getContributorStatus()))
+            ->end()
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param ArticleInterface $object
+     */
+    public function prePersist($object)
+    {
+        $this->sort($object);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configureFormFields(FormMapper $formMapper)
+    {
+        $fragmentClass = $this->fragmentClass;
+        $subject = $this->getSubject();
+
+        $formMapper
+            ->with('General', array('class' => 'col-md-8'))
+                ->add('title', 'text', array(
```

Comment:
Used for text, textarea, choice & datetimepicker only. it seems to be more complicated for model_autocomplete et collection in this fragment case